### PR TITLE
store: add tests to ensure `store/tikv` has no dependency on `parser`

### PR DESCRIPTION
**Pull request by breeswish**
Previously we have a PR that removes such dependency: https://github.com/pingcap/tidb/pull/5834

Now this PR further adds a unit test to ensure that no dependency will be introduced in future.

@siddontang PTAL

### CODE REVIEW

File: store/tikv/tikv_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	"strings"
+
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/parser"
+)
+
+type testTiKVSuite struct {
+}
+
+var _ = Suite(&testTiKVSuite{})
```
Comment (tiancaiamao) [inline store/tikv/tikv_test.go]: I'm not sure this test really runs, no `TestT` and no `SetUpSuite`, please check it.

File: store/tikv/tikv_test.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	if pkgName == "C" {
+		return
+	}
+	if _, exists := (*pkgs)[pkgName]; exists {
+		return
+	}
+	if strings.HasPrefix(pkgName, "golang_org") {
+		pkgName = path.Join("vendor", pkgName)
+	}
+	pkg, err := build.Import(pkgName, srcDir, 0)
```
[the remark this answers is not in the document]
Comment (breeswish) [reply store/tikv/tikv_test.go]: means supplying no additional flags

File: store/tikv/tikv_test.go (hunk 1 above)
Comment (breeswish) [reply store/tikv/tikv_test.go]: Confirmed, it runs:

1. checked out 1 commit before PR #5834 
2. git cherry-pick this commit
3. `make test`:

```
...
FAIL: tikv_test.go:51: testTiKVSuite.TestParserNoDep

tikv_test.go:64:
    c.Assert(importName == parserPkgName, IsFalse,
        Commentf("`%s` is imported from `%s`, which is a child dependency of `%s`", parserPkgName, pkgName, currentPkgName))
... obtained bool = true
... `github.com/pingcap/tidb/parser` is imported from `github.com/pingcap/tidb/table/tables`, which is a child dependency of `github.com/pingcap/tidb/store/tikv`
...
```

**Comment by tiancaiamao** [comment]
LGTM

**Comment by shenli** [comment]
@siddontang PTAL

**Pull request merged**